### PR TITLE
C.2.16,17,18 Sensei mentorship contract notes (view, create, edit)

### PR DESCRIPTION
**Pull request by natalieloke**
# Feature

Use Case: 

C.2.16 | C | C.2 | View Student Notes
C.2.17 | C | C.2 | Create a Student Note
C.2.18 | C | C.2 | Edit a Student Note

Feature: Mentorship Tools

# Changelog:
- see commit log

## Checklist:

- [x] Merged latest develop
- [x] PR title makes sense

## Screenshots (where relevant):
<a href="https://www.loom.com/share/ba7aaef8713a430880505c7205af92af">
    <p>Digi Dojo | Notes - View, add and edit - Watch Video</p>
    
  </a>
  
<a href="https://www.loom.com/share/82832e0c5d0a4bdaad0238b4d6b038a6">
    <p>Digi Dojo | Empty Notes - Watch Video</p>
    
  </a>

### CODE REVIEW

File: src/components/Mentorship/Notes/index.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+                        <Button onClick={() => onEditNote(note)}>Edit this note</Button>
+                      ) : (
+                        <Button onClick={() => onReset()}>Cancel editing this note</Button>
+                      )}
+                    </div>
+                  </Panel>
+                )
+              })}
+            </Collapse>
+          )}
```
Comment (natalieloke) [inline src/components/Mentorship/Notes/index.js]: do note that i wasn't able to extract this into it's own local component. I tried doing that but it caused the expanded note to close each time i clicked on 'Edit this note' button (if u watch the video).

File: src/components/Mentorship/Notes/index.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+            <Empty description="No notes yet" />
+          ) : (
+            <Collapse ghost>
+              {map(notes, note => {
+                return (
+                  <Panel header={note.title} key={note.key}>
+                    <p>{note.body}</p>
+                    <div>
+                      <div className="row">
+                        <small className="text-muted text-uppercase">
```
Comment (rawrrencee) [inline src/components/Mentorship/Notes/index.js]: `row` without `col-12`
image

Comment (rawrrencee) [reply src/components/Mentorship/Notes/index.js]: not sure if adding col will adjust the margin but i think would b nice if the words were aligned

File: src/components/Mentorship/Notes/index.js [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+      </div>
+    )
+  }
+
+  return (
+    <Card title="Notes">
+      <div className="row d-flex align-items-center justify-content-center mb-md-4">
+        <div
+          className="col-12 col-md-4"
+          style={{ paddingRight: '20px', borderRight: '1px solid #e8e8e8' }}
```
Comment (rawrrencee) [inline src/components/Mentorship/Notes/index.js]: is there mobile responsive for this? not sure how borderRight looks like on mobile.
but anyway, due to time constraints, we can say Tasks/Notes are ideally used on Desktop. so put a disclaimer. coz Tasks like mafan to make responsive, idk if i have the time.... prolly won't..

File: src/components/Mentorship/Notes/index.js [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+  const getNotes = async () => {
+    const response = await getAllNotes(id)
+    if (response && !isNil(response.notes)) {
+      setNotes(sortDescAndKeyNoteId(response.notes))
+    }
+  }
+
+  useEffect(() => {
+    getNotes()
+    forceUpdate({}) // required to disable 'submit' button of the form at the beginning
```
Comment (rawrrencee) [inline src/components/Mentorship/Notes/index.js:41]: quite curious, how does this work/why is this needed for this form as compared to other forms? ie whats the difference that makes this a need o: interesting

File: src/components/Mentorship/Notes/index.js [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+          className="col-12 col-md-4"
+          style={{ paddingRight: '20px', borderRight: '1px solid #e8e8e8' }}
+        >
+          <NoteFormComponent />
+        </div>
+        <div className="col-12 col-md-8">
+          {isEmpty(notes) ? (
+            <Empty description="No notes yet" />
+          ) : (
+            <Collapse ghost>
```
Comment (rawrrencee) [inline src/components/Mentorship/Notes/index.js:148]: I might wrap this component in a PaginationWrapper if i have the time. So that if there's more than 10 (default set in constants) then it will not show all (so the page not so long)

**Review by rawrrencee — DISMISSED** [review]
chirpchirp

the usability is quite good, can refer to other notes while editing the current one. if i were doing it, it would be the standard modal method. nice touch.
left some comments to be answered but other than that i think its ok to go ahead~

File: src/components/Mentorship/Notes/index.js (hunk 1 above)
Comment (rawrrencee) [reply src/components/Mentorship/Notes/index.js]: this wasn't happening in the video right? coz i rewatched, didn't see it happening hahah so assuming its just a statement and not recorded

Comment (natalieloke) [reply src/components/Mentorship/Notes/index.js]: oh yea the video shows the correct one where the note still remains in the expanded form when you click on the 'Edit' button lol

Comment (natalieloke) [reply src/components/Mentorship/Notes/index.js]: was just describing what would have happened if i put this chunk into a local component!

File: src/components/Mentorship/Notes/index.js (hunk 4 above)
Comment (natalieloke) [reply src/components/Mentorship/Notes/index.js:41]: the difference is that the 'Add New Form' button is disabled right at the beginning! I made it such that it's enabled only when the form fields change

Comment (natalieloke) [reply src/components/Mentorship/Notes/index.js:41]: here's the antd example: https://ant.design/components/form/

ss of the form component i referenced
Screenshot 2021-04-15 at 3 32 47 PM

File: src/components/Mentorship/Notes/index.js (hunk 3 above)
Comment (natalieloke) [reply src/components/Mentorship/Notes/index.js]: Screenshot 2021-04-15 at 3 35 01 PM
^ is this ok?

Comment (rawrrencee) [reply src/components/Mentorship/Notes/index.js]: haha ok can for now, dun rlly like the borderRight appearing for mobile, if I got time do you mind I see if I can do anything abt it?

File: src/components/Mentorship/Notes/index.js (hunk 2 above)
Comment (natalieloke) [reply src/components/Mentorship/Notes/index.js]: updated alignment
Screenshot 2021-04-15 at 3 42 20 PM

File: src/components/Mentorship/Notes/index.js (hunk 3 above)
Comment (natalieloke) [reply src/components/Mentorship/Notes/index.js]: sure, thanks!

**Review by rawrrencee — APPROVED** [review]

File: src/components/Mentorship/Notes/index.js [hunk 6]
```diff
@@ -138,10 +138,7 @@ const NotesComponent = () => {
   return (
     <Card title="Notes">
       <div className="row d-flex align-items-center justify-content-center mb-md-4">
-        <div
-          className="col-12 col-md-4"
-          style={{ paddingRight: '20px', borderRight: '1px solid #e8e8e8' }}
-        >
```
Comment (natalieloke) [inline src/components/Mentorship/Notes/index.js]: i realised paddingRight here did nothing and border right could be the className also, hence the change
